brianchandotcom/liferay-portal
# Feature: Adding help text to API GUI

# Problem 

"/o/api"

Users are unable to understand how to use the search and filter parameters in the API effectively.

# Solution  

1. Integrated the Learn Resource to provide help text directly in the API GUI.
2. Wrapped the swagger-ui-react library with custom component to display help text alongside relevant fields.
3. Developed a new API to make the Learn Resource easily accessible for users.

# Verification Steps  

1. Navigate to /o/api.
2. Open any API that includes a search or filter parameter.
3. Observe a question mark icon near the input field. Hovering over it reveals a tooltip with helpful text and a link to the relevant documentation.

### Note  

1. Since `headless-discovery-web.json` is not available, refer to the following steps to run it locally:  [Liferay Learn Resources](https://github.com/liferay/liferay-portal/tree/master/learn-resources).
2. To add more help text to the GUI, update the JSON file (headless-discovery-web.json) with a valid key such as search, filter, sort, fields, nestedFields, or similar. These changes will automatically reflect in the GUI.
